src/content/blog/content-layer-deep-dive.mdx

```diff
@@ -0,0 +1,230 @@
+			// and instead add, delete or update entries as needed.
+			store.clear();
+
+			for (const item of feed.items) {
+				// The parseData helper uses the schema to validate and transform data
+				const data = await parseData({
+					id: item.guid,
+					data: item,
+				});
+
+				// The generateDigest helper lets you generate a digest based on the content. This is an optional
+				// optimization. When inserting data into the store, if the digest is provided then the store will
+				// check if the content has changed before updating the entry. This will avoid triggering a rebuild
+				// in development if the content has not changed.
+				const digest = generateDigest(data);
+
+				store.set({
+					id,
+					data,
+					// If the data source provides HTML, it can be set in the `rendered` property
+					// This will allow users to use the `<Content />` component in their pages to render the HTML.
+					rendered: {
+						html: data.description ?? '',
+					},
+					digest,
+				});
+			}
+		},
+	};
+}
+```
+
+## When _not_ to use content collections
+
+Previously it was clear when content collections were a good idea, because it was whenever you were using local content in your pages. The content layer API makes that less clear, because you can use it for any content source including live APIs. The key consideration when deciding is that the data is only updated when the site is built. This means it's perfect for prerendered pages, but also for [on-demand rendering (SSR)](https://docs.astro.build/en/guides/server-side-rendering/) where the data changes relatively rarely. A blog would be a good example - if you're writing a blog that's hosted in a CMS, you can trigger a build with a webhook whenever you publish a new post. The same applies for e-commerce, where a build can be triggered when a product is edited. If you are ok with waiting the time it takes to deploy the site to see updates, then content collections are a good choice. You will get the best performance and great developer experience.
+
+If you need your pages to update in near realtime or with personalized content, then you are better off using an on-demand rendering adapter, ideally with CDN cache headers to ensure that page loads are super fast. You can even combine the two using [server islands](https://5-0-0-beta.docs.astro.build/en/guides/server-islands) and get the best of both worlds - render the main content using content collections, and use a server island for real-time or personalized content.
+
+## What's next
+
+The Content Layer API is a big step forward for Astro, but we're just getting started. Current the data store is just a key value store, with limited filtering. This is fast but not memory efficient, and querying is not very flexibile. Our goal is to introduce a backend based on Astro DB in a future version, which should help it scale to hundreds of thousands of pages. It will also allow us to support more complex queries, and perhaps even real-time updates.
```

Comment:
```suggestion
The Content Layer API is a big step forward for Astro, but we're just getting started. Currently the data store is just a key value store, with limited filtering. This is fast but not memory efficient, and querying is not very flexibile. Our goal is to introduce a backend based on Astro DB in a future version to help it scale to hundreds of thousands of pages. It will also allow us to support more complex queries, and perhaps even real-time updates.
```